### PR TITLE
Use (but limit) system setting for results per page in package management grid #modxbughunt

### LINKED ISSUE #12518: Default Per Page number of packages listed not related to the Default Per Page System Setting
The System Settings has a preference called Default Per Page (default_per_page), described as “The default number of results to show in grids throughout the manager”, but this number is not being used as the default number of packages listed per page in the manager.

**Extras > Installer > Package Management > Packages > Per Page = 10**

Instead it is just set as 10 in here:

**Path:** manager/assets/modext/workspace/package.grid.js
**Line 86:** `,pageSize: 10`

### PR DESCRIPTION
**Pull request by mkschell**
Fixes #12518 #modxbughunt

Package management results per page was hard-coded to 10. The bug report suggested using the system setting `default_per_page` but there could be performance problems if this number was really high (since that grid may cause remote requests out to the package provider). So a default cap of 25 was applied to the system setting in this case.

### CODE REVIEW

**Comment by OptimusCrime** [comment]
You introduce a new variable here that is not used as far as I can see. 

Also, instead of using the number class you can write the logic as

```
Math.min(parseInt(MODx.config.default_per_page), 25);
```

Other than that, good work :)

**Comment by Mark-H** [comment]
Test failure is due to a XML request timeout, not related to the PR.

**Comment by netProphET** [comment]
@OptimusCrime I had removed the variable originally, then added it back in to impose the limit. Unless you're referring to the introduction of another variable?

Thanks for the suggestion of Math.min(). Committed.

**Comment by OptimusCrime** [comment]
Looks good, but you should remove the variable here: https://github.com/modxcms/revolution/pull/13348/files#diff-283367202eada3aed53c62d14f3604ddR77

**Comment by rtripault** [comment]
Just wanted to link to #13088

**Comment by opengeek** [comment]
@netProphET I believe they are referring to the definition of `var pageSize` which is never used.

File: manager/assets/modext/workspace/package.grid.js [hunk 1]
```diff
@@ -74,6 +74,7 @@ MODx.grid.Package = function(config) {
         };
     }
 
+    var pageSize = parseInt(MODx.config.default_per_page)
```
Comment (opengeek) [inline manager/assets/modext/workspace/package.grid.js]: I don't believe this is used anywhere.

**Pull request merged**